drivers/serial/Kconfig.gecko

```diff
@@ -10,6 +10,6 @@ config UART_GECKO
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
 	select SOC_GECKO_USART
-	select PINCTRL if SOC_GECKO_SERIES1
+	select PINCTRL if SOC_FAMILY_SILABS_S1
```

Comment:
(unrelated to change) but why is pinctrl only selected for a specific series and not the rest?

Reply:
I think `PINCTRL` is also needed for Series 2. For Series 2, the symbol is set in defconfigs of the boards. Obviously, this does not make sense. I believe this symbol should relocate in `soc/silabs/s*/Kconfig.soc`.